### ISSUE TITLE
`UnboundLocalError` in `postprocessing.coverage()` for BND calls kills the whole task and BNDs get incorrect `END`

### ISSUE THREAD

**Issue by awgymer** [fritzsedlazeck/Sniffles#611]
In `v2.8.0`

## Summary

In `postprocessing.coverage()` the local variable `end` is not set for BND calls
([`src/sniffles/postprocessing.py:84-92`](https://github.com/fritzsedlazeck/Sniffles/blob/7fcaf8671e211d57632ace8ae713a1b2e0801656/src/sniffles/postprocessing.py#L84-L92)):

```python
for svcall in calls:
    start = svcall.pos
    if svcall.svtype == "INS":
        end = start + 1
    elif svcall.svtype == "BND":
        if svcall.bnd_info.is_first:
            start -= 1
    else:
        end = svcall.pos + abs(svcall.svlen)
```

BND calls then go into the same block as INS, and that block reads `end`
([`src/sniffles/postprocessing.py:104`](https://github.com/fritzsedlazeck/Sniffles/blob/7fcaf8671e211d57632ace8ae713a1b2e0801656/src/sniffles/postprocessing.py#L104) and [`src/sniffles/postprocessing.py:126`](https://github.com/fritzsedlazeck/Sniffles/blob/7fcaf8671e211d57632ace8ae713a1b2e0801656/src/sniffles/postprocessing.py#L126)):

```python
    if svcall.svtype in ("INS", "BND"):
        ...
        try:
            svcall.coverage_end = int(cv[end + config.coverage_binsize])
        except IndexError:
            ...
```

Two things go wrong:

1. If no call earlier in the same list has set `end`, this raises `UnboundLocalError`. The handler only catches `IndexError`, so the exception goes up to the worker. `Task.call_candidates` fails, and **every SV of that task is lost**, not only the BND.
2. If an earlier call in the same list did set `end`, there is no exception, but the BND silently takes `coverage_end` and `coverage_downstream` from the **end position of the previous call**. `sv.ALL_TYPES` puts BND last, so any INS, DEL, DUP or INV call in the same task hides the crash and gives the BND a wrong coverage value.

So issue 1 appears only in tasks where a BND is the first call to reach the coverage step, in every other case you hit issue 2. 

## Suggested correction

Give the BND branch an explicit `end`, in the same way as INS:

```python
    elif svcall.svtype == "BND":
        end = start + 1
        if svcall.bnd_info.is_first:
            start -= 1
```

`end` should also be set/reset inside the loop before the `if`, so that a value can never leak from the previous call.

**Comment by hermannromanek** [comment]
Hi @awgymer 

Thank you for the report as well as your analysis and the suggestion - I included it, so this bug will be fixed in the upcoming version 2.8.1

Thanks & best regards,
Hermann

**Comment by lfpaulin** [comment]
Solved in v2.8.1, if error persist please re-open
